src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py

```diff
@@ -0,0 +1,41 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+#pylint: skip-file
+import unittest
+import mock
+
+from azure.cli.core._util import CLIError
+from azure.cli.command_modules.acs.custom import (
+    _validate_service_principal,
+    _build_service_principal
+)
+
+class spClient:
+    def __init__(self, spMock, appMock):
+        self.service_principals=spMock
+        self.applications=appMock
+
+class AcsServicePrincipalTest(unittest.TestCase):
+    def test_validate_service_principal_ok(self):
+        spMock = mock.Mock()
+        spMock.list.return_value = []
+        client = spClient(spMock, None)
+        _validate_service_principal(client, '27497b5e-7ea6-4ff2-a883-b3db4e08d937')
+
+        self.assertTrue(spMock.list.called)
+        spMock.list.assert_has_calls([mock.call(filter="servicePrincipalNames/any(c:c eq '27497b5e-7ea6-4ff2-a883-b3db4e08d937')")])
+
+
+    def test_validate_service_principal_fail(self):
+        spMock = mock.Mock()
+        spMock.list.side_effect=KeyError('foo')
+        client = spClient(spMock, None)
+        try:
+            _validate_service_principal(client, '27497b5e-7ea6-4ff2-a883-b3db4e08d937')
```

Comment:
Using [assertRaise](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) is a better approach to test expected exception here.